api/src/main/java/io/grpc/NameResolver.java

```diff
@@ -652,11 +676,19 @@ public ConfigOrError getServiceConfig() {
 
     @Override
     public String toString() {
-      return MoreObjects.toStringHelper(this)
-          .add("addresses", addresses)
-          .add("attributes", attributes)
-          .add("serviceConfig", serviceConfig)
-          .toString();
+      ToStringHelper stringHelper = MoreObjects.toStringHelper(this);
+      if (addressesOrError.hasValue()) {
```

Comment:
Seems we should add a `toString()` to StatusOr.

Reply:
Done.